lib/streamlit/runtime/caching/cache_utils.py

```diff
@@ -80,6 +97,16 @@ def ttl_to_seconds(
         return math.inf
     if isinstance(ttl, timedelta):
         return ttl.total_seconds()
+    if isinstance(ttl, str):
+        parts = _TIMEDELTA_STR_RE.match(ttl)
```

Comment:
Oh, that's super cool! I didn't want to introduce a dep just for this, but since Pandas is already a deb and it does this, I'll just use your code. 🙏